src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/ConvertData.cpp

```diff
@@ -2,89 +2,124 @@
 
 #include "complex/DataStructure/DataArray.hpp"
 #include "complex/DataStructure/DataGroup.hpp"
+#include "complex/Utilities/ParallelDataAlgorithm.hpp"
 
 using namespace complex;
 
 namespace Detail
 {
 template <typename O, typename D>
-void ConvertData(DataArray<O>& inputArray, DataArray<D>& outputArray)
+class ConvertDataAlg
 {
-  size_t size = inputArray.getSize();
-  for(size_t v = 0; v < size; ++v)
+public:
+  using InputStoreType = AbstractDataStore<O>;
+  using OutputStoreType = AbstractDataStore<D>;
+
+  ConvertDataAlg(InputStoreType& inputStore, OutputStoreType& outputStore)
+  : m_InputStore(inputStore)
+  , m_OutputStore(outputStore)
   {
-    if constexpr(std::is_same<O, D>::value)
-    {
-      // inputArray and destination arrays have the same type
-      outputArray[v] = inputArray[v];
-    }
-    else if constexpr(std::is_same<O, bool>::value)
-    {
-      // inputArray array is a boolean array
-      outputArray[v] = (inputArray[v] ? 1 : 0);
-    }
-    else if constexpr(std::is_same<D, bool>::value)
-    {
-      // Destination array is a boolean array
-      outputArray[v] = (inputArray[v] != 0);
-    }
-    else
+  }
+
+  void convert(size_t start, size_t end) const
+  {
+    // std::cout << "\tConvertDataAlg: " << start << " to " << end << std::endl;
+
+    for(size_t v = start; v < end; ++v)
     {
-      // All other cases
-      outputArray[v] = static_cast<D>(inputArray[v]);
+      // std::cout << "\tConvertDataAlg: index " << v << std::endl;
+
+      if constexpr(std::is_same<O, D>::value)
+      {
+        // inputArray and destination arrays have the same type
+        m_OutputStore.setValue(v, m_InputStore.getValue(v));
+      }
+      else if constexpr(std::is_same<O, bool>::value)
+      {
+        // inputArray array is a boolean array
+        m_OutputStore.setValue(v, (m_InputStore.getValue(v) ? 1 : 0));
+      }
+      else if constexpr(std::is_same<D, bool>::value)
+      {
+        // Destination array is a boolean array
+        m_OutputStore.setValue(v, (m_InputStore.getValue(v) != 0));
+      }
+      else
+      {
+        // All other cases
+        m_OutputStore.setValue(v, static_cast<D>(m_InputStore.getValue(v)));
+      }
     }
   }
-}
+
+  /**
+   * @brief operator () This is called from the TBB stye of code
+   * @param range The range to compute the values
+   */
+  void operator()(const Range& range) const
+  {
+    convert(range.min(), range.max());
+  }
+
+private:
+  InputStoreType& m_InputStore;
+  OutputStoreType& m_OutputStore;
+};
 
 template <typename T>
 Result<> ConvertData(DataStructure& dataStructure, const ConvertDataInputValues* inputValues)
 {
   DataArray<T>& inputArray = dataStructure.getDataRefAs<DataArray<T>>(inputValues->SelectedArrayPath);
+  AbstractDataStore<T>& inputStore = inputArray.getDataStoreRef();
+
+  ParallelDataAlgorithm dataAlg;
+  dataAlg.setRange(0, inputArray.size());
+  dataAlg.setParallelizationEnabled(false);
```

Comment:
Check if parallelization disabled is intended.

Reply:
@mmarineBlueQuartz Can we check if the DataArray is out-of-core then turn off parallel otherwise leave it on?